packages/design-system/src/components/popup/index.js

```diff
@@ -35,76 +36,84 @@ import { themeHelpers } from '../../theme';
 import { noop } from '../../utils';
 import { getTransforms, getOffset } from './utils';
 import { PLACEMENT } from './constants';
-
-// TODO scrollbar update, commented out until design updates are done
-const DEFAULT_POPUP_Z_INDEX = 11;
-
+const DEFAULT_TOPOFFSET = 32;
 const Container = styled.div.attrs(
   ({
     $offset: { x, y, width, height },
     fillWidth,
     fillHeight,
     placement,
-    zIndex,
+    isRTL,
+    invisible,
   }) => ({
     style: {
-      transform: `translate(${x}px, ${y}px) ${getTransforms(placement)}`,
+      transform: `translate(${x}px, ${y}px) ${getTransforms(placement, isRTL)}`,
       ...(fillWidth ? { width: `${width}px` } : {}),
       ...(fillHeight ? { height: `${height}px` } : {}),
-      zIndex,
+      ...(invisible ? { visibility: `hidden` } : {}),
     },
   })
 )`
   /*! @noflip */
   left: 0px;
   top: 0px;
   position: fixed;
-
+  z-index: 2;
+  max-height: ${({ topOffset }) => `calc(100vh - ${topOffset}px)`};
   ${themeHelpers.scrollbarCSS};
 `;
-
 function Popup({
+  isRTL = false,
   anchor,
   dock,
   children,
   renderContents,
   placement = PLACEMENT.BOTTOM,
-  zIndex = DEFAULT_POPUP_Z_INDEX,
   spacing,
   isOpen,
+  invisible,
   fillWidth = false,
   fillHeight = false,
   onPositionUpdate = noop,
   refCallback = noop,
+  topOffset = DEFAULT_TOPOFFSET,
```

Comment:
Where is `DEFAULT_TOPOFFSET  = 32` coming from? The admin bar?

A sane default here is `0`. Because in a non-WordPress standalone fullscreen editor there won't be any offset.

The `32` should be passed from WordPress only

Reply:
oh ok will change it to 0